### DISCUSSION TITLE
favorites page

### DISCUSSION THREAD

**Discussion by DorisW3** [neuefische/questions#698] [Web] [answered]
hi,

I need help with  displaying all favorites (pictures the user can like or unlike) on their own page, but I couldn't make it dynamic (the toggle of the favorite button doesn't work, but the onClick does!). 

I checked the discussion board, asked other people and looked at recap projects...

here you can find my pull request: 
https://github.com/DorisW3/capstone-project/pull/14

Thank you!

**Comment by Kheirah** [comment, marked as answer]
For the button you need to pass the id of the image.
```js
<FavoriteButton onToggleFavorite={() => onToggleFavorite(picture.id)} />
```
For the function itself you need to accept that id.
_Additionally, I changed `pictures` to `images` because you want the most recent state._
```js
function handleToggleFavorite(id) {
  setImages(
    images.map((image) =>
      image.id === id
        ? { ...image, isFavorite: !image.isFavorite }
        : image
    )
  );
```
I also left these comments on your PR. These suggestions may not work out of the box because I haven't run the code. But it should give you the idea how to implement it.

**Comment by DorisW3** [comment]
Vielen lieben Dank, durch die Vorgeschlagenen Änderungen funktioniert es jetzt!! :gratitude-danke:
